scheduling/dispatcher.rb

```diff
@@ -65,11 +71,17 @@ def start_strand(strand)
 
   def start_cohort
     scan.each do |strand|
+      break if shutting_down
       @notifiers << start_strand(strand)
     end
   end
 
   def wait_cohort
+    if shutting_down
+      Clog.emit("Shutting down.") { {unfinished_strand_count: @notifiers.length} }
+      Kernel.exit! if @notifiers.empty?
```

Comment:
Likely regular exit is okay here, the exit! is for dire situations (like, this process has to die or else mutex might not work, per apoptosis).

The difference is one will unwind the stack because it is implemented as an exception that can be rescued.

On the other hand, we don't actually have anything interesting in the stack, so the exit syscall (which `Kernel.exit!` is) is pretty harmless, and the downside risk of some Bad Code rescuing the exception is much less than mutual exclusion not working (since you only get here when notifiers.empty? is true)

Reply:
Thanks. Change to use regular `exit`.